common/logging/log4j2/src/main/java/io/apiman/common/logging/log4j2/Log4j2LoggerFactory.java

```diff
@@ -17,28 +17,73 @@
 
 import io.apiman.common.logging.IApimanLogger;
 import io.apiman.common.logging.IDelegateFactory;
+import io.apiman.common.logging.LogLevel;
 import io.apiman.common.logging.annotations.ApimanLoggerFactory;
 
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.logging.log4j.Level;
 import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.core.LoggerContext;
 import org.apache.logging.log4j.message.FormattedMessageFactory;
 
+;
+
 /**
  * Log4j2 logger factory.
  *
  * @author Marc Savy {@literal <msavy@redhat.com>}
  */
 @ApimanLoggerFactory("log4j2")
 public class Log4j2LoggerFactory implements IDelegateFactory {
-
     private final FormattedMessageFactory formattedMessageFactory =  new FormattedMessageFactory();
+    private final LoggerContext context = (LoggerContext) LogManager.getContext(false);
 
     @Override
     public IApimanLogger createLogger(String name) {
-        return new Log4j2LoggerImpl(LogManager.getLogger(name, formattedMessageFactory));
+        return new Log4j2LoggerImpl(context.getLogger(name, formattedMessageFactory));
     }
 
     @Override
     public IApimanLogger createLogger(Class<?> klazz) {
-        return new Log4j2LoggerImpl(LogManager.getLogger(klazz, formattedMessageFactory));
+        return new Log4j2LoggerImpl(context.getLogger(klazz, formattedMessageFactory));
+    }
+
+    @Override
+    public IDelegateFactory overrideLoggerConfig(File newLoggerConfig) {
+        context.setConfigLocation(newLoggerConfig.toURI());
+        return this;
+    }
+
+    @Override
+    public IDelegateFactory overrideLoggerConfig(Map<String, LogLevel> newLoggerConfig) {
```

Comment:
Remember to create issues to implement this for each different logger impl where possible